### ISSUE TITLE
Support for wildcard domains in EDNS

### ISSUE THREAD

**Issue by pmilot** [F5Networks/k8s-bigip-ctlr#2813]
#### Title
Support for wildcard domains in EDNS

#### Description
Ability to support wildcard characters in EDNS.  
For example: *.alpha.mydomain.net

#### Actual Problem
Currently VS and tlsProfiles support wildcard characters but not EDNS.

I need the ability to create a single vip sending traffic for all subdomains to an ingress controller and would like the wildcard domain to be created by EDNS.

Currently EDNS rejects any wildcard characters.

The ExternalDNS "alpha-ingress" is invalid: spec.domainName: Invalid value: "*.alpha.mydomain.net": spec.domainName in body should match '^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$'

```
---
apiVersion: "cis.f5.com/v1"
kind: VirtualServer
metadata:
  namespace: kube-system
  name: alpha-ingress
  labels:
    f5cr: "true"
spec:
  host: "*.alpha.mydomain.net"
  virtualServerName: alpha-ingress-tls
  virtualServerAddress: 10.xxx.xxx.181
  tlsProfileName: alpha-ingress-tls
  snat: auto
  iRules: 
    - /Common/Shared/k8s_ingress_sni_irule
  pools:
  - monitor:
      type: tcp
      interval: 3
      timeout: 10
    path: /
    service: rke2-ingress-nginx-controller
    servicePort: 443
---
 apiVersion: cis.f5.com/v1
 kind: TLSProfile
 metadata:
   namespace: kube-system
   name: alpha-ingress-tls
   labels:
     f5cr: "true"
 spec:
   tls:
     termination: reencrypt
     serverSSL: /Common/serverssl
     clientSSL: /Common/Shared/k8s_ingress_clientside_tls_profile
     reference: bigip            # --> reference profiles created in BIG-IP by User
   hosts:
   - "*.alpha.mydomain.net"
---
apiVersion: "cis.f5.com/v1"
kind: ExternalDNS
metadata:
  name: alpha-ingress
  labels:
    f5cr: "true"
spec:
  domainName: "*.alpha.mydomain.net"
  dnsRecordType: A
  loadBalanceMethod: round-robin
  pools:
  - dnsRecordType: A
    loadBalanceMethod: round-robin
    dataServerName: /Common/Lab-BIP-BIGIP-16.1.3-quad-001
    monitor:
      type: tcp
      interval: 5
      timeout: 16
```

**Comment by pmilot** [comment]
@trinaths I'm not sure what other information you would require.

The VS (LTM) objects are being created correctly with the wildcard.  I would like the GTM WideIP to get configured with the A record for *.alpha.mydomain.net same as when using a FQDN.

This is what it would look like when I create it manually.  However currently the EDNS CRD does not accept any '*' character.

```
gtm wideip a \*.alpha.mydomain.net {
    pools {
        \*.alpha.mydomain.net {
            order 0
        }
    }
}
gtm pool a \*.alpha.mydomain.net_pool {
    members {
        Lab-BIP-BIGIP-16.1.3-quad-001:/alpha-cluster/Shared/alpha_ingress_tls_443 {
            member-order 0
        }
    }
}
```

**Comment by trinaths** [comment]
Created [CONTCNTR-3917] for internal tracking.

**Comment by charanm08** [comment]
Added in this PR : https://github.com/F5Networks/k8s-bigip-ctlr/pull/2962. Will be available in 2.14 release

**Comment by charanm08** [comment]
Added in this PR : https://github.com/F5Networks/k8s-bigip-ctlr/pull/2962. Will be available in 2.14 release
